### ISSUE TITLE
[Bug] ambiguous column error when building off a model with `date_<grain>` field

### ISSUE THREAD

**Issue by fivetran-jamie** [dbt-labs/dbt_metrics#103]
### Is this a new bug in dbt_metrics?

- [X] I believe this is a new bug in dbt_metrics
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

We are currently investigating adding metrics to the dbt Ad Reporting [package](https://github.com/fivetran/dbt_ad_reporting), in which each end model has a column named `date_day`. After defining a metric like this
```yml
metrics:
  - name: spend
    label: Ad spend (Fivetran)
    model: ref('ad_reporting__ad_report')

    type: sum
    sql: spend
    description: Total spend (in currency of individual platforms)

    timestamp: date_day
    time_grains: [day, week, month]

    dimensions:
      - platform
      - campaign_id
      - campaign_name
      - ad_group_id
      - ad_group_name
      - ad_id
      - ad_name
      - account_id
      - account_name
```

When I attempt to run a model calling `metrics.calculate` on this, I receive an error about `date_day` being an ambiguous column name, as it's also a column name in the helper calendar model. In some under-the-hood macros in this package, `date_<grain>` uses dot notation, while in some places, it's not prepended with a CTE or model alias, which is where the error is coming from I believe

### Expected Behavior

I'm able to build metrics off of a model with a `date_<grain>` column name (pretty standard)

The alternative option is to not support metrics on models with these sorts of column names

### Steps To Reproduce

1. Build a model with a `date_day` field (or install Ad Reporting/an individual Ad Package) 
2. Define a metric using that model
3. Call metric.calculate on that model 
4. dbt run

### Relevant log output

```shell
Database Error in model test_metrics (models/test_metrics.sql)
21:50:28    Column name date_day is ambiguous at [139:17]
21:50:28    compiled SQL at target/run/jamie_dbt_dev/models/test_metrics
```

### Environment

```markdown
- dbt-adapter & version: dbt-bigquery 1.2.0
- dbt_metrics version: >0.3.0, <= 0.4.0
```

### Which database adapter are you using with dbt?

bigquery

**Comment by callum-mcdata** [comment]
@fivetran-jamie thanks for filing this issue! This was previously found and addressed in PR #98 , which will be part of the v0.4.0 release later this month.

In the meantime, would it be possible to test out a dummy timestamp field that has the same values as date_day but is instead called `date_day__test`? Then before actual release of the ad reporting package with metric definitions this would be resolved.

**Comment by fivetran-jamie** [comment]
oh great! sorry didn't PRs. and yep `date_day_test` gets around the issue
